inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -195,6 +194,20 @@ def __init__(self,
                  params: Dict) -> None:
         super().__init__(dispatcher_broker, repo, params)
 
+    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
+        # get the GUID from the system using FW tool
+        extracted_guids = extract_guids(
+            self._fw_tool, ["System Firmware type", "system-firmware type"])
+        if manifest_guid:
+            if manifest_guid not in extracted_guids:
+                raise FotaError(
+                    f"GUID in manifest does not match any system firmware GUID on the system")
+            else:
+                return manifest_guid
+        else:
+            return extracted_guids[0]
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Consider checking if 'extracted_guids' is not empty before accessing its elements to avoid potential IndexError.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
        # get the GUID from the system using FW tool
        extracted_guids = extract_guids(
            self._fw_tool, ["System Firmware type", "system-firmware type"])
        if not extracted_guids:
            raise FotaError("No system firmware GUIDs found.")
        if manifest_guid:
            if manifest_guid not in extracted_guids:
                raise FotaError(
                    f"GUID in manifest does not match any system firmware GUID on the system")
            else:
                return manifest_guid
        else:
            return extracted_guids[0]
```